### ISSUE TITLE
import third party libraries (pouchDB plugins) in unit test throw an error

### ISSUE THREAD

**Issue by EugeneSnihovsky** [marcoturi/ionic2-boilerplate#18]
Hi,

I'm use pouchDB (has typings) with [pouchdb-quick-search](https://github.com/nolanlawson/pouchdb-quick-search) plugin (has no typings).

In **declarations.d.ts** I add:
```typescript
declare module 'pouchdb-quick-search'
```
also tried
```typescript
declare module 'pouchdb-quick-search' {
    const plugin: PouchDB.Plugin;
    export = plugin;
}
```

Then in service:
```typescript
import * as PouchDB from 'pouchdb';
import * as quickSearch from 'pouchdb-quick-search';
PouchDB.plugin(quickSearch); // this line throw error. If I comment it - all works, but I need this plugin
```

Both variant works well in developer (server) and production (apk) mode, but unit test won't start and throw error
![error image](https://sc-cdn.scaleengine.net/i/ad50dc84c7fb6577eb67fa05d008043c.png)
When I run dev server, pouchdb-quick-search plugin look like
![plugin object in console image](https://sc-cdn.scaleengine.net/i/ef21211cbd03bdd7cc82e61d7972e908.png)

Other third party libraries with and without typings works fine. Not work only pouchDB plugins (pouchdb-quick-search, [pouchdb-find](https://github.com/nolanlawson/pouchdb-find), etc).

And what is interesting, I can't even **console.log(quickSearch)** in test. All references to quickSearch throw an error. Also maybe help: I can't even test component, that use mocked pouchDB service with pouchdb-quick-search, it also throw same error.
```typescript
{provide: PouchDbService, useClass: PouchDbServiceMock},
```
I would be glad of any help

**Comment by marcoturi** [comment]
I never used pouchdb-quick-search but i tried pouchdb. So this is how I'm mocking/testing parts with it, hope it helps.

```
export class MockPouchDB {
    dbName: string;
    opt;
    db;

    constructor(dbName, opt) {
        this.dbName = dbName;
        this.opt = opt;
        this.db = [];
    }

    destroy(dbName) {
        const index = this.db.indexOf(dbName);
        if (index) {
            this.db.splice(index, 1);
        }
    }

    get(documentId) {
        if (this.db[this.dbName] && this.db[this.dbName]._id === documentId) {
            return Promise.resolve(this.db[this.dbName]);
        } else {
            return Promise.reject({name: 'not_found'});
        }
    }

    put(data) {
        if (!data._id) {
            throw new Error('Missing id');
        }
        if (!data._rev) {
            data._rev = 0;
        }
        data._rev = parseInt(data._rev, 10) + 1;
        data._rev = data._rev.toString();
        this.db[this.dbName] = data;
        return Promise.resolve(data);
    }
}

```

And here the mock/test of a service:

 ```
   beforeEach(() => {
        spyOn(MyService.prototype, 'initDB').and.callFake(() => {
            return true;
        });
        myService = new MyService();
        myService['db'] = new MockPouchDB(APP_CONST.DB.NAME, {auto_compaction: true});
    });
```

So I basically suggest you to wrap you call in method and spy/callFake it. 

A last tip. For native JavaScript classes (i.e. new Audio/Media/AudioContext) I use this method to mock/test them, maybe it could be useful in your case:

```
    beforeEach(() => {
        window.Audio = () => {
            return new MockAudio();
        };
        webAudioService = new WebAudioService('');
    });
```

**Comment by EugeneSnihovsky** [comment]
Hi again. Thanks for your response. This error appear during compilation. I found a [problem place](https://github.com/calvinmetcalf/lie/blob/master/lib/index.js#L13). I don't know is it webpack or PhantomJS or lie bug, but when I locally in node_modules wrap **if (!process.browser)** construction in try/catch - problems gone. This is a bad solution (6 modules in my project requires lie), but I need fast fix. Also I created pull request for lie module fix.
